src/components/molecules/NavList.tsx

```diff
@@ -1,53 +1,187 @@
-import React from 'react';
-import {List, ListItemButton, ListItemText, ListSubheader} from '@mui/material';
+import React, { useState } from "react";
+import {
+  List,
+  ListItemButton,
+  ListItemText,
+  ListSubheader,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+
+const commonStyle = {
+  fontWeight: "bold",
+  bgcolor: "black",
+  color: "white",
+  fontSize: 20,
+};
+
+function generateButtonStyle(isSelected: boolean) {
+  return {
+    ...(isSelected && {
+      backgroundColor: "lightgrey",
+      padding: "1px 16px",
+      borderRadius: "0 24px 24px 0",
+      "&:hover": {
+        backgroundColor: "lightgrey",
+      },
+    }),
+  };
+}
+
+function NavList() {
+  const [selectedLink, setSelectedLink] = useState<string | null>(null);
+
+  const handleLinkClick = (link: string) => {
+    setSelectedLink(link);
+  };
+
+  const isLinkSelected = (link: string) => selectedLink === link;
 
-function NavList (){
   return (
     <List
-      sx={{ width: '100%', maxWidth: 360, bgcolor: 'background.paper' }}
+      sx={{ width: "100%", height: "100%", maxWidth: 360, ...commonStyle }}
       component="nav"
     >
-      {/* TODO: routing on click and show current selection*/}
-      <ListItemButton>
-        <ListItemText primary="Recent Work" />
+      <ListItemButton
+        component={Link}
+        to="/"
+        sx={{ ...generateButtonStyle(isLinkSelected("/")) }}
+        onClick={() => handleLinkClick("/")}
+      >
+        <ListItemText
+          primary="Recent Work"
+          sx={{
+            color: isLinkSelected("/") ? "black" : "white",
+            padding: "1px",
+          }}
+        />
+      </ListItemButton>
+      <ListItemButton
+        component={Link}
+        to="/contact"
+        sx={{ ...generateButtonStyle(isLinkSelected("/contact")) }}
+        onClick={() => handleLinkClick("/contact")}
+      >
+        <ListItemText
+          primary="Contact"
+          sx={{
+            color: isLinkSelected("/contact") ? "black" : "white",
+            padding: "1px",
+          }}
+        />
       </ListItemButton>
-      <ListItemButton>
-        <ListItemText primary="About" />
+      <ListItemButton
+        component={Link}
+        to="/about"
+        sx={{ ...generateButtonStyle(isLinkSelected("/about")) }}
+        onClick={() => handleLinkClick("/about")}
+      >
+        <ListItemText
+          primary="About"
+          sx={{
+            color: isLinkSelected("/about") ? "black" : "white",
+            padding: "1px",
+          }}
+        />
       </ListItemButton>
-      <ListItemButton>
-        <ListItemText primary="Contact" />
+      <ListItemButton
+        component={Link}
+        to="/video-list-page"
+        sx={{ ...generateButtonStyle(isLinkSelected("/video-list-page")) }}
+        onClick={() => handleLinkClick("/video-list-page")}
+      >
+        <ListItemText
```

Comment:
Won't be an actual nav item, but the video list will be the actual rendered component for most other routes